include/experimental/__p1673_bits/blas3_matrix_rank_k_update.hpp

```diff
@@ -149,31 +149,16 @@ void symmetric_matrix_rank_k_update(
 {
   using size_type = std::common_type_t<SizeType_A, SizeType_C>;
 
-  if constexpr (std::is_same_v<Triangle, upper_triangle_t>) {
-    // [A_00  A_01  A_02]   [A_00            ]
-    // [      A_11  A_12] * [A_01  A_11      ]
-    // [            A_22]   [A_02  A_12  A_22]
-    for (size_type row = 0; row < C.extent(0); ++row) {
-      for (size_type col = 0; col < C.extent(1); ++col) {
-        const size_type lower = row < col ? col : row; // max
-        const size_type upper = A.extent(0);
-        for (size_type k = lower; k < upper; ++k) {
-          C(row, col) += alpha * A(row, k) * A(col, k);
-        }
-      }
-    }
-  }
-  else {
-    // [A_00            ]   [A_00  A_10  A_20]
-    // [A_10  A_11      ] * [      A_11  A_21]
-    // [A_20  A_21  A_22]   [            A_22]
-    for (size_type row = 0; row < C.extent(0); ++row) {
-      for (size_type col = 0; col < C.extent(1); ++col) {
-        const size_type lower = 0;
-        const size_type upper = row < col ? row : col; // min
-        for (size_type k = lower; k < upper; ++k) {
-          C(row, col) += alpha * A(row, k) * A(col, k);
-        }
+  constexpr bool lower_tri =
+    std::is_same_v<Triangle, lower_triangle_t>;
+  using size_type = std::common_type_t<SizeType_A, SizeType_C>;
```

Comment:
Line 154 is redundant with line 150.

```suggestion
  constexpr bool lower_tri =
    std::is_same_v<Triangle, lower_triangle_t>;
```